### PR TITLE
fix(radar)(route): Pixiv search & Radar rules

### LINKED ISSUE #7699: pixiv radar-rules 更新
pixiv页面的链接规则已改变，对应的radar-rules需要修改。

#### 相关文档地址

- https://docs.rsshub.app/social-media.html#pixiv

- 额外信息（日志、报错等）

页面|之前URL|现在URL
-|-|-
Pixiv用户收藏|/bookmark.php&id=11|/users/11/bookmarks/artworks
Pixiv用户动态|/member.php&id=11|/users/11
Pixiv排行榜|/search.php|/tags/:keyword

### PR DESCRIPTION
**Pull request by yindaheng98**
## 该 PR 相关 Issue / Involved issue

Close #7699

并修改了排行榜和搜索的路由使之兼容新的Radar rules

## 完整路由地址 / Example for the proposed route(s)

```routes
/pixiv/ranking/daily
/pixiv/ranking/male
/pixiv/ranking/female
/pixiv/ranking/original
/pixiv/ranking/rookie
/pixiv/search/fgo
/pixiv/search/fgo/date
/pixiv/search/fgo/null/r18
/pixiv/search/fgo/null/safe
```

## 新RSS检查列表 / New RSS Script Checklist

- [ ] 这个PR中包含了新的路由吗? Does this PR add new route?
  - 如果有, 请完成检查列表. If yes, please finish the check list
  - **如果你的PR符合下方某个事项, 也请注明. If any of the checklist item meets your PR, please fill it out.**
  - [x] <- 这样打勾
- [x] 是否提供了文档? Documentation provided?
  - [x] 是否提供了英文文档? EN Documentation provided?
- [ ] 是否支持全文获取? Is this RSS Script support fulltext?
  - [ ] 如果全文获取中需要访问文章链接, 是否使用了缓存? If fulltext requires to fetch detail pages, is cache used in the process?
  - [缓存说明](https://docs.rsshub.app/joinus/quick-start.html#ti-jiao-xin-de-rsshub-gui-ze-bian-xie-jiao-ben-shi-yong-huan-cun) | [How to use cache](https://docs.rsshub.app/joinus/quick-start.html#ti-jiao-xin-de-rsshub-gui-ze-bian-xie-jiao-ben-shi-yong-huan-cun)
- [ ] 目标是否有明显的反爬/频率限制? Is there any sign of anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? (延长缓存时间, 写文档说明, etc.) If yes, do your code reflect this sign? (e.g. write documentations, use long cache time)
- [ ] 目标是否有提供日期？ Is there a date in the source?
  - [ ] 如果有，包是否正确解析？ If there is, can this script provide this info?
  - [ ] 如果有提供解析能力，时区是否正确调整？ Is the timezone correctly provided?
  - 如果有提供日期，但是没有提供解析，请说明原因 If there is a date but this script does not parse, please provide your reason.
- [ ] 是否引入的新的包? Any new package introduced?
  - 如果有, 请说明原因. If yes, please state your reason
- [ ] 是否使用了`Puppeteer`? Make use of `Puppeteer`?
  - 如果有, 请说明原因. If yes, please state your reason

### CODE REVIEW

File: docs/en/social-media.md [hunk 1]
```diff
@@ -209,7 +209,7 @@ These feed do not include boosts (a.k.a. reblogs). RSSHub provides a feed for us
 
 ### Keyword
 
-<RouteEn author="DIYgod" example="/pixiv/search/麻衣/popular/2" path="/pixiv/search/:keyword/:order?/:r18?" :paramsDesc="['keyword', 'rank mode, empty or other for time order, popular for popular order', 'filte R18 content, 0 to no filter, 1 to only not R18, 2 to only R18, default to 0']" radar="1" rssbud="1"/>
+<RouteEn author="DIYgod" example="/pixiv/search/麻衣/popular/2" path="/pixiv/search/:keyword/:order?/:mode?" :paramsDesc="['keyword', 'rank mode, empty or other for time order, popular for popular order', 'filte R18 content, safe or 1 to only not R18, r18 or 2 to only R18, empty or other o no filter']" radar="1" rssbud="1"/>
```
Comment (SettingDust) [inline docs/en/social-media.md:212]: `empty or other o no filter` o? to?

empty or others to disable filter

File: docs/social-media.md [hunk 2]
```diff
@@ -519,7 +519,7 @@ Tiny Tiny RSS 会给所有 iframe 元素添加 `sandbox="allow-scripts"` 属性
 
 ### 关键词
 
-<Route author="DIYgod" example="/pixiv/search/麻衣/popular/2" path="/pixiv/search/:keyword/:order?/:r18?" :paramsDesc="['关键词', '排序方式，popular 按热门度排序，空或其他任意值按时间排序', '过滤 R18 内容，0 为不过滤，1 为只看非 R18 内容，2 为只看 R18 内容，默认为 0']" radar="1" rssbud="1"/>
+<Route author="DIYgod" example="/pixiv/search/麻衣/popular/2" path="/pixiv/search/:keyword/:order?/:mode?" :paramsDesc="['关键词', '排序方式，popular 按热门度排序，空或其他任意值按时间排序', '过滤 R18 内容，safe 或 1 为只看非 R18 内容，r18 或 2 为只看 R18 内容，空其他任意值为不过滤']" radar="1" rssbud="1"/>
```
Comment (SettingDust) [inline docs/social-media.md:522]: `空其他任意值为不过滤`
留空或其他值将不过滤

另外，这里和上面的英文文档建议把这两个（空不需要）做成表格，并且只保留 safe, r18, empty 在文档中，没必要在文档中保留0, 1, 2

File: lib/routes/pixiv/ranking.js [hunk 3]
```diff
@@ -32,12 +32,27 @@ const links = {
     week_r18g: 'https://www.pixiv.net/ranking.php?mode=r18g',
 };
 
+const alias = {
+    daily: 'day',
+    weekly: 'week',
+    monthly: 'month',
+    male: 'day_male',
+    female: 'female',
```
Comment (SettingDust) [inline lib/routes/pixiv/ranking.js:40]: male 和 female 行为不一致？

**Review by SettingDust — CHANGES_REQUESTED** [review]

File: lib/routes/pixiv/ranking.js (hunk 3 above)
Comment (yindaheng98) [reply lib/routes/pixiv/ranking.js]: 是bug，已修复

**Review by yindaheng98 — COMMENTED** [review]
reviewed

**Review by SettingDust — APPROVED** [review]

**Pull request merged**